analysis/quantifier-comparisons/13-spatial-transcriptomics-benchmarking.Rmd

```diff
@@ -0,0 +1,411 @@
+}
+
+all_spe_filter <- all_spe_list %>%
+  purrr::map(filter_spe)
+```
+
+When we look at our results, we will also want to visualize them so we will make a custom function to plot the results.
+Because of the bug that is present in `SpatialExperiment::read10XVisium()` function, we will make two separate functions, one for spaceranger only spes and one for alevin-fry + spaceranger spes.
+
+```{r}
+# custom function for plotting spe results and coloring by column of colData of choice
+plot_fry_spe <- function(spe, sample, column){
+  # plot spots only 
+  p1 <- ggspavis::plotSpots(spe, 
+                            x_coord = "pxl_col_in_fullres", 
+                            y_coord = "pxl_row_in_fullres", 
+                            annotate = column) +
+    scale_color_viridis_c()
+  
+  # plot with tissue underneath
+  p2 <- ggspavis::plotVisium(spe, 
+                                   x_coord = "pxl_col_in_fullres", 
+                                   y_coord = "pxl_row_in_fullres", 
+                                   fill = column) +
+    scale_fill_viridis_c()
+  
+  # arrange plots and add sample name as title 
+  grid.arrange(p1, p2, nrow = 1, top = grid::textGrob(sample))
+}
+
+
+plot_spaceranger_spe <- function(spe, sample, column){
+  # plot spots only 
+  p1 <- ggspavis::plotSpots(spe, 
+                            # switch the order of row and col to flip the image 
+                            x_coord = "pxl_row_in_fullres", 
+                            y_coord = "pxl_col_in_fullres", 
+                            annotate = column) +
+    scale_color_viridis_c() 
+  
+  # plot with tissue underneath
+  p2 <- ggspavis::plotVisium(spe, 
+                                   x_coord = "pxl_row_in_fullres", 
+                                   y_coord = "pxl_col_in_fullres", 
+                                   fill = column) +
+    scale_fill_viridis_c()
+  
+  # arrange plots and add sample name as title 
+  grid.arrange(p1, p2, nrow = 1, top = grid::textGrob(sample))
+  
+}
+```
+
+First we will look at the per cell metrics: mitochondrial reads per cell, total UMI per cell, and total genes detected per cell. 
+
+```{r}
+# % mitochondrial reads/ spot 
+ggplot(coldata_df_common, aes(x = tool, y = subsets_mito_percent, fill = tool)) + 
+  geom_boxplot() +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("Mito Percent") + 
+  xlab("")
+```
+```{r}
+plot_fry_spe(all_spe_filter$`alevin-fry`, "alevin-fry", "subsets_mito_percent")
+```
+```{r}
+plot_spaceranger_spe(all_spe_filter$spaceranger, "spaceranger", "subsets_mito_percent")
+```
+Overall it looks like mitochondrial content is low and fairly similar across both tools. 
+
+
+```{r}
+# total UMI/ spot 
+ggplot(coldata_df_common, aes(x = sum, color = tool)) + 
+  geom_density() + 
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("UMI/spot") + 
+  xlab("")
```

Comment:
The UMI count difference here (and genes detected below) seems pretty significant, but I'm not sure what to make of it. (and thank you for using density plots!)